docs/TempUG.md

```diff
@@ -545,93 +546,118 @@ Please use show lessonIndex before deleting task!
 ```
 
 #### For Schedule:
 <!-- use "-subject SUBJECTS" for lesson -->
-Format: `command COMPULSORY [optional]` (for list specific format)
-* Format info 1
-* Format info 2
-
+Format: `editLesson [INDEX] [-name NAME] [-start TIME] [-end TIME] [-day DATE] [-subject SUBJECT]` (for list specific format)
+* Edits the lesson at the specified `INDEX` if it is provided, otherwise edits the currently/lastly shown lesson.
+* You must specify at least one field to edit.
+* The name of the lesson after editing cannot be the same (case-insensitive) as other existing lessons in the schedule list.
+* The start time of the lesson after editing cannot be after the end time of the lesson, if the end time is specified, and vice versa.
+* The lesson cannot clash (same day, start and end all specified and are overlapping) in time with other lessons  in the schedule list.
+* Please note that unlike the `addLesson` command, the `editLesson` command's "-subject" flag only accepts one subject, not multiple subjects (parameter is SUBJECT, not SUBJECTS).
+* Refer to the parameter constraints [here](#parameter-summary).
 <box type="tip" seamless>
 
 **Tips:**
-- Tip 1
-- Tip 2
+- If you are currently in ___SCHEDULE list___, the command can be shortened to `edit`.
 
 </box>
 
 Example usages:
-* `some code here`
-* `another code here`
+* `editLesson 1 -start 12:35 -end 14:35 -day 2023/11/21 -subject English`
+* To edit the currently shown lesson in ___SCHEDULE list___:
+   * `edit -start 12:30 -end 14:30 -day 2023/11/20 -subject English`
 
 Success outputs:
-* Input: `code with compulsory parameters`
-* Input: `code with compulsory and optional parameters`
+* Input: `editLesson 1 -start 12:35 -end 14:35 -subject English`
 ```
-This block of code is for success outputs
+Edit success.
+from: Lesson lesson1 from 12:30 PM to 2:30 PM on 20-11-2023 for MATHEMATICS
+to: Lesson lesson1 from 12:35 PM to 2:35 PM on 20-11-2023 for ENGLISH
 ```
 Failure outputs:
-* Input: `invalid command code here`
-  * Error: Explanation and solution here, this is because the flag has an incorrect value, bla bla bla
+* Input: `editLesson 1`
+  * Error: No field to edit given. Need to enter at least one field to edit.
 ```
-Invalid command with the error message here
+No edit detected. Please edit at least one field: name, start, end, subject, day to different value.
+```
+Failure outputs:
+* Input: `editLesson -name a new name` (assuming no lesson is currently shown)
+  * Error: No lesson shown. Show a lesson with the `show` command.
+```
+Using edit command without specifying index when no entry is shown. 
+Usage: edit [INDEX] (at least one of unique [-name|subject|day|start|end VALUE]). 
+For example, edit 1 -name lesson2 -subject English -day 23/12 -start 14:30 -end 16:30
+If you want to edit the currently shown lesson, you could omit the index. 
+Note your edited 'name' must not already in the schedule and 'start' must be before 'end'.
+```
+Failure outputs:
```

Comment:
Same as above.

Reply:
Thank you for pointing that out. Required update made.